src/ray/gcs/gcs_server/gcs_placement_group_scheduler.cc

```diff
@@ -195,16 +83,32 @@ void GcsPlacementGroupScheduler::ScheduleUnplacedBundles(
     return;
   }
 
+  RAY_CHECK(bundles.size() == selected_nodes.size());
+
+  // Covert to a map of bundle to node.
+  ScheduleMap bundle_to_node;
+  for (size_t i = 0; i < selected_nodes.size(); ++i) {
+    bundle_to_node[bundles[i]->BundleId()] =
+        NodeID::FromBinary(selected_nodes[i].Binary());
+  }
+
   auto lease_status_tracker =
-      std::make_shared<LeaseStatusTracker>(placement_group, bundles, selected_nodes);
+      std::make_shared<LeaseStatusTracker>(placement_group, bundles, bundle_to_node);
   RAY_CHECK(placement_group_leasing_in_progress_
                 .emplace(placement_group->GetPlacementGroupID(), lease_status_tracker)
                 .second);
 
-  const auto &pending_bundles = GetUnplacedBundlesPerNode(bundles, selected_nodes);
-  for (const auto &node_to_bundles : pending_bundles) {
-    const auto &node_id = node_to_bundles.first;
-    const auto &bundles_per_node = node_to_bundles.second;
+  // Covert to a set of bundle specifications grouped by the node.
```

Comment:
To be clear:
These 5 lines are the function of original `GetUnplacedBundlesPerNode`.
It is too simple and no need to abstract to a method.